src/accounts/superfluid.ts

```diff
@@ -0,0 +1,197 @@
+        if (!this.alephx) throw new Error("Account is not initialized");
+        const flow = await this.alephx.getFlow({
+            sender: this.address,
+            receiver,
+            providerOrSigner: this.wallet.provider,
+        });
+        // check wrapped balance, if none throw
+        const balance = ethers.BigNumber.from(
+            await this.alephx.balanceOf({ account: this.address, providerOrSigner: this.wallet.provider }),
+        );
+        if (balance.lt(this.alephToWei(alephPerHour))) {
+            throw new Error("Not enough ALEPHx to increase flow");
+        }
+        if (!flow) {
+            const op = this.alephx.createFlow({
+                sender: this.address,
+                receiver,
+                flowRate: this.alephPerHourToFlowRate(alephPerHour).toString(),
+            });
+            await op.exec(this.wallet.provider.getSigner());
+        } else {
+            const newFlowRate = ethers.BigNumber.from(flow.flowRate.toString()).add(
+                this.alephPerHourToFlowRate(alephPerHour),
+            );
+            const op = this.alephx.updateFlow({
+                sender: this.address,
+                receiver,
+                flowRate: newFlowRate.toString(),
+            });
+            await op.exec(this.wallet.provider.getSigner());
+        }
+    }
+
+    public async decreaseALEPHxFlow(receiver: string, alephPerHour: Decimal): Promise<void> {
+        if (!this.wallet) throw Error("PublicKey Error: No providers are set up");
+        if (!this.alephx) throw new Error("Account not initialized");
+        const flow = await this.alephx.getFlow({
+            sender: this.address,
+            receiver,
+            providerOrSigner: this.wallet.provider,
+        });
+        if (!flow) {
+            throw new Error("Flow does not exist");
+        } else {
+            const newFlowRate = ethers.BigNumber.from(flow.flowRate.toString()).sub(
+                this.alephPerHourToFlowRate(alephPerHour),
+            );
+            if (newFlowRate.lte(0)) {
+                const op = this.alephx.deleteFlow({
+                    sender: this.address,
+                    receiver,
+                });
+                await op.exec(this.wallet.provider.getSigner());
+            } else {
+                const op = this.alephx.updateFlow({
+                    sender: this.address,
+                    receiver,
+                    flowRate: newFlowRate.toString(),
+                });
+                await op.exec(this.wallet.provider.getSigner());
+            }
+        }
+    }
+}
+
+export function createFromAvalancheAccount(account: AvalancheAccount, rpc?: string): SuperfluidAccount {
+    if (account.wallet) {
+        return new SuperfluidAccount(account.wallet, account.address, account.publicKey);
+    }
+    throw new Error("Wallet is required");
+    // @todo: implement for node.js
+    if (!rpc) throw new Error("RPC or Provider is required");
```

Comment:
```suggestion
    if (!rpc) throw new Error("RPC or Wallet is required");
```

Reply:
Provider can be something else in this context. When using a node.js environment, one also does not usually speak of a wallet (which are the GUIs for users)